services/terrain/src/terrain/clients/ezid.clj

```diff
@@ -0,0 +1,84 @@
+(ns ^{:author psarando} terrain.clients.ezid
+  (:use [ring.util.http-response :only [charset]]
+        [slingshot.slingshot :only [try+ throw+]])
+  (:require [cemerick.url :as curl]
+            [clj-http.client :as http]
+            [clojure.string :as string]
+            [terrain.util.config :as config]))
+
+(defn- auth-params
+  []
+  [(config/ezid-username) (config/ezid-password)])
+
+(defn- ezid-uri
+  [& components]
+  (str (apply curl/url (config/ezid-base-url) components)))
+
+(defn- anvl-escape
+  [value]
+  (-> value
+      (string/replace #"%"  "%25")
+      (string/replace #"\n" "%0A")
+      (string/replace #"\r" "%0D")
+      (string/replace #":"  "%3A")))
+
+(defn- anvl-unescape
+  [value]
+  (-> value
+      (string/replace #"%3A" ":")
+      (string/replace #"%0D" "\r")
+      (string/replace #"%0A" "\n")
+      (string/replace #"%25" "%")))
```

Comment:
This might be a good idea, except we still need to guarantee order, since we don't want to escape everything else before attempting to escape `%`. So maybe a vector of vectors could work instead of a map?